Sorry, the test suite has a habit of hanging rather than failing nicely when there is an issue in the `sphinx_agent` code which can make things harder to diagnose.

Checking your PR out locally and running `hatch test -ipy=3.14 -isphinx=7 -- tests/sphinx-agent/ -x` to exit on the first failure I get
```
        try:
>           result = await self.protocol.send_request_async("sphinx/build", params)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E           pygls.exceptions.JsonRpcInternalError: sphinx-build failed: sphinx.errors.ExtensionError: Handler <function update_symbols at 0x7fb03be49c70> for event 'source-read' threw an exception (exception: 'PosixPath' object has no attribute 'endswith')
```

--- a/lib/esbonio/esbonio/server/features/sphinx_manager/config.py
+++ b/lib/esbonio/esbonio/server/features/sphinx_manager/config.py
@@ -427,6 +427,9 @@ def sphinx_config_struct_hooks(field_name: str):
         if isinstance(obj, list):
             return typ(command=obj)
 
+        if isinstance(obj, str):
+            return typ(command=[obj])
+
         return typ(**obj)
 
     return _structure_list_or_subprocess

--- a/lib/esbonio/esbonio/server/features/sphinx_manager/config.py
+++ b/lib/esbonio/esbonio/server/features/sphinx_manager/config.py
@@ -389,7 +389,11 @@ def register_structure_hooks(converter: cattrs.Converter):
         for a in attrs.fields(SphinxConfig)
     }
 
-    hook = cattrs.gen.make_dict_structure_fn(SphinxConfig, converter, **fields)
+    hook = cattrs.gen.make_dict_structure_fn(
+        SphinxConfig,
+        converter,
+        **fields,  # type: ignore[arg-type]
+    )
 
     def _structure_sphinx_config(obj, typ):
         """A wrapper around the ``cattrs.gen`` hook to implement multiple names mapping

--- a/lib/esbonio/esbonio/sphinx_agent/app.py
+++ b/lib/esbonio/esbonio/sphinx_agent/app.py
@@ -6,6 +6,8 @@ import pathlib
 import sys
 import typing
 
+from packaging.version import Version
+from sphinx import __version__ as __sphinx_version__
 from sphinx.application import Sphinx as _Sphinx
 from sphinx.errors import ThemeError
 from sphinx.util import console
@@ -15,6 +17,8 @@ from sphinx.util.logging import NAMESPACE as SPHINX_LOG_NAMESPACE
 from . import types
 from .database import Database
 from .log import DiagnosticFilter
+
+_SPHINX_9 = Version(__sphinx_version__) >= Version("9.0.0")
 
 if typing.TYPE_CHECKING:
     from typing import IO
@@ -35,9 +39,12 @@ logger = sphinx_logger.getChild("esbonio")
 sphinx_log_setup = sphinx_logging_module.setup
 
 
-def setup_logging(app: Sphinx, status: IO, warning: IO):
+def setup_logging(app: Sphinx, status: IO, warning: IO, verbosity=None):
     # Run the usual setup
-    sphinx_log_setup(app, status, warning)
+    if _SPHINX_9:
+        sphinx_log_setup(app, status, warning, verbosity=verbosity)
+    else:
+        sphinx_log_setup(app, status, warning)
 
     # Attach our diagnostic filter to the warning handler.
     for handler in sphinx_logger.handlers:

--- a/lib/esbonio/esbonio/sphinx_agent/config.py
+++ b/lib/esbonio/esbonio/sphinx_agent/config.py
@@ -113,7 +113,7 @@ class SphinxConfig:
         # Sphinx 8.1 changed the way arguments are passed to the `Sphinx` class.
         # See: https://github.com/swyddfa/esbonio/issues/912
         if len(values) == 0:
-            sphinx_args = m_Sphinx.call_args.kwargs
+            sphinx_args = dict(m_Sphinx.call_args.kwargs)
 
         if sphinx_args is None:
             return None

